common.mk

```diff
@@ -53,9 +53,10 @@ $(VERILOG_FILE) $(SMEMS_CONF) $(TOP_ANNO) $(TOP_FIR) $(sim_top_blackboxes): $(FI
 	cd $(base_dir) && $(SBT) "project tapeout" "runMain barstools.tapeout.transforms.GenerateTop -o $(VERILOG_FILE) -i $(FIRRTL_FILE) --syn-top $(TOP) --harness-top $(MODEL) -faf $(ANNO_FILE) -tsaof $(TOP_ANNO) -tsf $(TOP_FIR) $(REPL_SEQ_MEM) -td $(build_dir)"
 	cp $(build_dir)/firrtl_black_box_resource_files.f $(sim_top_blackboxes)
 
-$(HARNESS_FILE) $(HARNESS_ANNO) $(HARNESS_FIR) $(sim_harness_blackboxes): $(FIRRTL_FILE) $(ANNO_FILE) $(sim_top_blackboxes)
+# note: this depends on sim_top_blackboxes to avoid race condition where both GenerateX calls are done at the same time
+$(HARNESS_FILE) $(HARNESS_SMEMS_CONF) $(HARNESS_ANNO) $(HARNESS_FIR) $(sim_harness_blackboxes): $(FIRRTL_FILE) $(ANNO_FILE) $(sim_top_blackboxes)
 	cd $(base_dir) && $(SBT) "project tapeout" "runMain barstools.tapeout.transforms.GenerateHarness -o $(HARNESS_FILE) -i $(FIRRTL_FILE) --syn-top $(TOP) --harness-top $(VLOG_MODEL) -faf $(ANNO_FILE) -thaof $(HARNESS_ANNO) -thf $(HARNESS_FIR) $(HARNESS_REPL_SEQ_MEM) -td $(build_dir)"
-	grep -v ".*\.cc" $(build_dir)/firrtl_black_box_resource_files.f > $(sim_harness_blackboxes)
+	cp $(build_dir)/firrtl_black_box_resource_files.f $(sim_harness_blackboxes)
```

Comment:
Ok so this is actually an issue. Currently... with the grep it removes any `.cc` files that conflict with Rocket Chip so that the simulator won't complain about name collision. Before the `.*\.cc` the grep just removed `SimSerial` `DTM` `JTAG`. However, I don't think adding a grep here is the right thing to do. I think we should remove the additions in the `GenerateSimFiles` step. What was the initial reasoning in the `GenerateSimFiles` step for adding things like... `SimDTM` `SimJTAG`, etc @grebe

Reply:
Do we still have that makefile bit that makes sure the two blackbox `.f` files are unique? If so then the only issue with the `Sim` components is that `emulator.cc` includes all of the possible interfaces and so if the config elaborated doesn't include all interfaces then you will not be able to build a simulator.

Reply:
Ahh. I was unaware of that since I was running VCS sim. Hmm. The simple approach is to "dedup" the two `.f` files... but I think that is something that needs to be cleaner in the end.

Reply:
Yeah IIRC there was some github code-golf on the deduping. Maybe its just in some other PR?

Reply:
Yes there was. I will update this PR with the changes.